microsoft/testsuites/performance/storageperf.py

```diff
@@ -210,92 +206,68 @@ def perf_storage_generic_fio_test(self,
+                    test_result=result,
+                    test_name=f"{size}_MB_4K",
+                    num_jobs=num_jobs,
+                    block_size=4,
+                    time=240,
+                    size_mb=size,
                 )
-
-                fio_result[test_name]["status"] = "Succeeded"
-                fio_result[test_name]["error"] = ""
-                fio_result[test_name]["metrics"] = {
-                    "qdepth": result.qdepth,
-                    "mode": result.mode,
-                    "iops": str(result.iops),
-                    "latency": str(result.latency),
-                    "iodepth": result.iodepth,
-                }
-
-            except Exception as e:
-                fio_result[test_name]["status"] = "Failed"
-                fio_result[test_name]["error"] = str(e)
-                fio_result[test_name]["metrics"] = {}
-                log.error(e)
-
-            rm.remove_directory(str(path.joinpath(f"{filename}*")), sudo=True)
-
-        log.debug(fio_result)
-
-        # Saving the result
-        report_path = node.get_working_path().joinpath("fio_report")
-        report_file = report_path.joinpath("fio_test_result.json")
-        if not os.path.exists(report_path):
-            os.mkdir(report_path)
-
-        with open(report_file, "w") as f:
-            f.write(json.dumps(fio_result))
+        else:
+            for testcase in testcases:
+                id = str(uuid.uuid4())
+                try:
+                    start_iodepth = testcase.get("start_iodepth", 1)
+                    max_iodepth = testcase.get("max_iodepth", 1)
+                    num_jobs = []
+                    iodepth_iter = start_iodepth
+                    core_count = node.tools[Lscpu].get_core_count()
+                    while iodepth_iter <= max_iodepth:
+                        num_jobs.append(min(iodepth_iter, core_count))
+                        iodepth_iter = iodepth_iter * 2
+
+                    time = testcase.get("time", 240)
+                    block_size = testcase.get("block_size", 4)
+                    size_mb = testcase.get("size_mb", 512)
+                    overwrite = testcase.get("overwrite", False)
+
+                    test_name = f"{size_mb}_MB_{block_size}K"
+                    log.debug(f"Executing the FIO testcase : {test_name}")
+
+                    perf_disk(
+                        node=node,
+                        start_iodepth=start_iodepth,
+                        max_iodepth=max_iodepth,
+                        filename=f"{size_mb}_MB_FIO_{id}",
+                        test_result=result,
+                        test_name=test_name,
+                        num_jobs=num_jobs,
+                        block_size=block_size,
+                        time=time,
+                        size_mb=size_mb,
+                        overwrite=overwrite,
+                    )
+                except Exception as e:
+                    log.error(e)
```

Comment:
We should let the exception be raised. Else the testcase would get into PASSED state. We can remove the try-except block.

Reply:
We need try block as we have list of testcases. I have put assert with latest commit to reflect correct status of lisa testcase.